tembo-cli/src/cmd/apply.rs

```diff
@@ -434,26 +439,69 @@ fn get_trunk_installs(
     vec_trunk_installs
 }
 
-pub fn get_instance_settings() -> Result<HashMap<String, InstanceSettings>, anyhow::Error> {
-    let mut file_path = FileUtils::get_current_working_dir();
-    file_path.push_str("/tembo.toml");
+fn overlay_to_instance_settings(overlay: OverlayInstanceSettings) -> InstanceSettings {
+    InstanceSettings {
+        environment: overlay.environment.unwrap_or("default_environment".to_string()),
+        instance_name: overlay.instance_name.unwrap_or("default_instance_name".to_string()),
+        cpu: overlay.cpu.unwrap_or("default_cpu".to_string()),
+        memory: overlay.memory.unwrap_or("default_memory".to_string()), 
+        storage: overlay.storage.unwrap_or("default_storage".to_string()),
+        replicas: overlay.replicas.unwrap_or(1), // Set a default value for replicas here
+        stack_type: overlay.stack_type.unwrap_or("default_stack_type".to_string()),
+        postgres_configurations: overlay.postgres_configurations,
+        extensions: overlay.extensions,
+        extra_domains_rw: overlay.extra_domains_rw,
+    }
+}
 
-    let contents = match fs::read_to_string(file_path.clone()) {
-        Ok(c) => c,
-        Err(e) => {
-            panic!("Couldn't read context file {}: {}", file_path, e);
-        }
-    };
+fn merge_settings(base: &InstanceSettings, overlay: OverlayInstanceSettings) -> InstanceSettings {
```

Comment:
We should take a look at this crate to help handle merging potentially - maybe @vrmiguel has opinions here too? https://crates.io/crates/serde-toml-merge